src/lib/sputter/net/port.cpp

```diff
@@ -0,0 +1,108 @@
+#include "port.h"
+
+#include <cstring>
+#include <iostream>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+using namespace sputter::net;
+
+UDPPort::UDPPort()
+    : socketHandle(-1) {
+}
+
+UDPPort::~UDPPort() {
+  close();
+}
+
+bool UDPPort::open(uint16_t port) {
+  socketHandle = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
+  if (socketHandle < 0) {
+    std::cerr << "Error: Failed to create socket\n";
+    return false;
+  }
+
+  // Store port address
+  sockaddr_in bindAddress;
+  memset(&bindAddress, 0, sizeof(bindAddress));
+  bindAddress.sin_family = AF_INET;
+  bindAddress.sin_addr.s_addr = htonl(INADDR_ANY);
+  bindAddress.sin_port = htons(port);
+  if (bind(socketHandle, (sockaddr *)&bindAddress, sizeof(bindAddress)) < 0) {
+    std::cerr << "Error: Failed to bind socket\n";
+    return false;
+  }
+
+  return true;
+}
+
+void UDPPort::close() {
+  if (socketHandle >= 0) {
+    ::close(socketHandle);
+    socketHandle = -1;
+  }
+}
+
+bool UDPPort::send(const void *data, int dataSize, const std::string &address, uint16_t port) {
+  if (socketHandle < 0) {
+    std::cerr << "Error: Socket is not open\n";
```

Comment:
FYI a newline char won't flush the `cerr` stream, so you may not actually get output after this line is called, depending on how much is already buffered.

I suggest `std::endl` instead for that reason, unless you're in some perf-sensitive code path and you explicitly want to avoid flushing.